models/weather_api_handler.py

```diff
@@ -0,0 +1,74 @@
+#!usr/bin/python3
+"""
+module to handle all incoming request for weather information
+"""
+
+import requests
+from models.exceptions import *
+import os
+
+
+class WeatherRequestHandler:
+    """handle API for weather"""
+
+    forecastUrl = "https://api.weatherapi.com/v1/forecast.json"
+    # futureUrl = "https://api.weatherapi.com/v1/future.json"
+    weatherQueryUrl = ""
+    params = {}
+    selectedDayDifferences = 0
+
+    def __init__(self, weatherRequestInformation):
+        """initialise WeatherRequestHandler"""
+        WeatherRequestHandler.selectedDayDifferences = weatherRequestInformation["selected_days_weather_api"]
+        self.create_weather_query_params(weatherRequestInformation)
+
+    @classmethod
+    def create_weather_forecast_query_params(cls, weatherRequestInformation):
+        """create weather query url if selected is less than 14 days"""
+        WeatherRequestHandler.weatherQueryUrl = WeatherRequestHandler.forecastUrl
+        WeatherRequestHandler.params = {
+            "key": os.environ.get('WEATHER_API_KEY'),
+            "q": weatherRequestInformation["place_address"],
+            "days": weatherRequestInformation["selected_days_weather_api"]
+        }
+
+    # @classmethod
+    # def create_weather_future_query_params(cls, weatherRequestInformation):
+    #     """create weather query url if selected is between 14 days and 300 days"""
+    #     WeatherRequestHandler.weatherQueryUrl = WeatherRequestHandler.futureUrl
+    #     WeatherRequestHandler.params = {
+    #         "key": os.environ.get('WEATHER_API_KEY'),
+    #         "q": weatherRequestInformation["place_address"],
+    #         "dt": weatherRequestInformation["selected_date_event_api"]
+    #     }
+
+    @classmethod
+    def create_weather_query_params(cls, weatherRequestInformation):
+        """create weather query url"""
+        if not os.environ.get('WEATHER_API_KEY'):
+            raise ServerEnvironVariablesNotSet("WEATHER_API_KEY")
+
```

Comment:
this should be in a seperate method (as outlined above) because create_weather_query_params is also setting_request_url.